superset-frontend/src/explore/components/controls/DateFilterControl/components/CalendarFrame.tsx

```diff
@@ -29,12 +29,15 @@ import {
   FrameComponentProps,
 } from '../types';
 
-export function CalendarFrame(props: FrameComponentProps) {
-  let calendarRange = PreviousCalendarWeek;
-  if (CALENDAR_RANGE_SET.has(props.value as CalendarRangeType)) {
-    calendarRange = props.value;
-  } else {
-    props.onChange(calendarRange);
+export function CalendarFrame({ onChange, value }: FrameComponentProps) {
+  useEffect(() => {
+    if (!CALENDAR_RANGE_SET.has(value as CalendarRangeType)) {
+      onChange(PreviousCalendarWeek);
+    }
+  }, [onChange, value]);
+
+  if (!CALENDAR_RANGE_SET.has(value as CalendarRangeType)) {
+    return null;
```

Comment:
This functionality looks new, why added it?

Reply:
The whole purpose of this PR is to change the "default value" approach because it was broken.
What I'm doing here is if the initial value isn't one of the calendar range options, I'm returning `null` so in the next render I'll get the correct value.